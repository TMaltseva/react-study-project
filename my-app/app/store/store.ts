import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import { api } from '../services/api';
import selectedItemsReducer from '../services/selectedItemsSlice';
import { createWrapper } from 'next-redux-wrapper';

export const makeStore = () =>
  configureStore({
    reducer: {
      [api.reducerPath]: api.reducer,
      selectedItems: selectedItemsReducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware),
  });

setupListeners(makeStore().dispatch);

export type RootState = ReturnType<ReturnType<typeof makeStore>['getState']>;
export type AppDispatch = ReturnType<typeof makeStore>['dispatch'];

export const wrapper = createWrapper(makeStore);

export const store = makeStore();
