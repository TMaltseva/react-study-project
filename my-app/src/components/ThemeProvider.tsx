import React, { useState, useEffect } from 'react';
import { ThemeContext, Theme } from '../services/themeContext';

export const ThemeProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const [theme, setTheme] = useState<Theme | null>(null);

  useEffect(() => {
    const savedTheme = (localStorage.getItem('theme') as Theme) || 'light';
    setTheme(savedTheme);
    document.body.className = `${savedTheme}`;
  }, []);

  useEffect(() => {
    if (theme !== null) {
      document.body.className = `${theme}`;
    }
  }, [theme]);

  const toggleTheme = () => {
    const newTheme = theme === 'light' ? 'dark' : 'light';
    setTheme(newTheme);
    localStorage.setItem('theme', newTheme);
  };

  if (theme === null) {
    return <div>Loading...</div>;
  }

  return <ThemeContext.Provider value={{ theme, toggleTheme }}>{children}</ThemeContext.Provider>;
};

export default ThemeProvider;
