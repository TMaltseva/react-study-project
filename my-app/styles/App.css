#root {
    padding: 10px;
    max-width: 1280px;
    margin: 20px auto;
    text-align: center;
  }
  
  .logo {
    height: 6em;
    padding: 1.5em;
    will-change: filter;
    transition: filter 300ms;
  }
  .logo:hover {
    filter: drop-shadow(0 0 2em #646cffaa);
  }
  .logo.react:hover {
    filter: drop-shadow(0 0 2em #61dafbaa);
  }
  
  @keyframes logo-spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
  
  @media (prefers-reduced-motion: no-preference) {
    a:nth-of-type(2) .logo {
      animation: logo-spin infinite 20s linear;
    }
  }
  
  .card {
    padding: 2em;
  }
  
  .read-the-docs {
    color: #888;
  }
  
  #root {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    width: 100vw;
    gap: 40px;
  }
  
  #root button {
    width: 150px;
    height: 50px;
  }
  
  .sections-wrapper {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  
  .top-section {
    display: flex;
    justify-content: center;
    gap: 10px;
    margin-bottom: 40px;
    flex-wrap: wrap;
    width: 50vw;
  }
  
  .top-section div {
    display: flex;
    gap: 10px;
    align-items: center;
  }
  
  .bottom-section {
    display: flex;
    justify-content: space-between;
    gap: 40px;
    margin-bottom: 40px;
  }
  
  .bottom-section:not(:has(.right-section)) {
    justify-content: center;
  }
  
  .bottom-section p {
    margin: 10px 0;
  }
  
  .bottom-section p:last-child {
    margin-bottom: 20px;
  }
  
  .bottom-section-element {
    display: flex;
    align-items: center;
    cursor: pointer;
  }
  
  .bottom-section-element input {
    cursor: pointer;
  }
  
  .bottom-section-element label {
    display: flex;
  }
  
  .details {
    padding: 20px;
    text-align: center;
    display: flex;
    flex-direction: column;
    align-items: flex-end;
  }
  
  .details h2,
  .details p {
    text-align: end !important;
  }
  
  .bottom-section-element h2 {
    font-size: 1.2em;
    text-align: start;
    cursor: pointer;
    margin-left: 10px;
  }
  
  .error-message {
    font-size: 60px;
    margin: auto 20px;
    margin-bottom: 40px;
  }
  
  .pagination {
    display: flex !important;
    justify-content: center;
    align-items: center;
    margin-top: 20px;
    gap: 15px;
  }
  
  .pagination button {
    margin: 0 5px;
    width: 70px !important;
  }
  
  .left-section {
    border-right: 1px solid #ccc;
    padding-right: 30px;
  }
  
  .left-section:has(+ .right-section:empty) {
    border-right: none;
  }
  
  .right-section {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    flex: 2;
  }
  
  .right-section button {
    margin-top: 10px;
  }
  
  .bottom-section-element h2 {
    padding: 5px;
  }
  
  .bottom-section-element h2:hover {
    color: #646cffaa;
  }
  
  body.light {
    background-color: #ccc;
    color: black;
  }
  
  body.light #search-input {
    border-color: #ccc;
  }
  
  body.dark {
    background-color: black;
    color: #ccc;
  }
  
  .flyout {
    flex-basis: 100%;
    justify-content: center;
    display: none;
  }
  
  .flyout.active {
    display: flex;
  }
  
  #__next {
    display: flex;
    justify-content: center;
    align-items: center;
  }
  
  div:has(.error-message) {
    display: flex;
    flex-direction: column;
  }
  
  @media (max-width: 768px) {
    .top-section,
    .top-section div {
      flex-direction: column;
      gap: 20px;
    }
  
    .throw-error-btn,
    .theme-btn {
      margin: 0 auto;
    }
  
    .bottom-section-element h2 {
      font-size: 1em;
    }
  }