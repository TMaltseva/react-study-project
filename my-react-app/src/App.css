#root {
  padding: 10px;
  max-width: 1280px;
  margin: 20px auto;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

#root {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100vw;
  gap: 40px;
}

#root button {
  width: 150px;
  font-size: 18px;
  height: 50px;
}


.form {
  margin-top: 2rem;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  gap: 1rem;
  border-radius: 5px;
  padding: 2rem;
}

.input {
  border: none;
  border-bottom: 0.1rem solid #e3e3e346;
  outline: none;
  padding: 1rem;
  min-width: 30rem;
  background-color: transparent;
}

.input::placeholder {
  text-align: start;
}

input[type='file'] {
  border: none;
}

.label {
  cursor: pointer;
  font-weight: 900;
}

.label:hover {
  color: #646cffaa;
}

.nav {
  padding: 1rem;
}

.nav_list {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 10%;
  list-style-type: none;
}

.nav_item a {
  text-decoration: none;
  text-transform: capitalize;
  color: #fff;
  font-size: 2rem;
  font-weight: 500;
}

a:hover {
  color: #646cffaa;
}

a.active {
  color: #646cffaa;
}

.select {
  border: none;
  border-bottom: 0.1rem solid #e3e3e346;
  outline: none;
  padding: 1rem;
  min-width: 32rem;
  background-color: transparent;
}

.uncontrolled,
.controlled {
  display: flex;
  justify-content: center;
  font-size: 22px;
  text-transform: capitalize;
}

.wrapper {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.wrapper:nth-of-type(3) {
  font-size: 18px;
  font-weight: normal;
}

.input_item {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 2.5rem;
  padding: 0 2rem 0 0;
}

.input_item #terms {
  min-width: 5rem;
}

.input_item > input[type='radio'] {
  min-width: 1rem;
}

.btn_wrapper {
  margin-top: 5rem;
  width: 100%;
}

.errors {
  position: relative;
  margin-top: -0.7rem;
  color: #646cffaa;
  align-self: end;
  text-transform: lowercase;
  max-width: 40rem;
  padding-right: 30px;
}

.main-page {
  max-width: 1024px;
  margin: 0 auto;
}

.tile-wrapper {
  margin-top: 3rem;
  display: flex;
  justify-content: center;
  gap: 2rem;
  flex-wrap: wrap;
}

.tile-style {
  padding: 2rem;
  border-radius: 1rem;
  width: 25rem;
}

.highlight {
  border: 0.5rem solid #646cffaa;
}

.no-border {
  border: none;
}
