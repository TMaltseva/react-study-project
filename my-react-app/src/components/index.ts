export { Button } from './Button/Button';
export { Form } from './Form/Form';
export { Input } from './Input/Input';
export { Label } from './Label/Label';
export { Nav } from './Nav/Nav';
export { Select } from './Select/Select';
