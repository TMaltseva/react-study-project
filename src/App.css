#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

#root {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100vw;
  gap: 40px;
}

#root button {
  width: 150px;
  height: 50px;
}

.top-section {
  display: flex;
  justify-content: center;
  gap: 10px;
  margin-bottom: 40px;
}

.top-section div {
  display: flex;
  gap: 10px;
  align-items: center;
}

.bottom-section {
  width: 90vw;
}

.bottom-section div {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  gap: 10px;
}

.bottom-section p {
  width: 200px;
  margin: 0 auto;
}

.bottom-section p:last-child {
  margin-bottom: 20px;
}

.bottom-section-element {
  border: 1px solid #ccc;
  border-radius: 10px;
  padding: 10px;
  text-align: center;
}

.bottom-section-element h2 {
  font-size: 1.2em;
}

#root div:first-child:not(.bottom-section-elements):not(.top-section):not(.top-section div) {
  display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: center;
}

.error-message {
  font-size: 60px;
  margin: auto 20px;
  margin-bottom: 40px;
}

@media (max-width: 768px) {
  .top-section,
  .top-section div {
    flex-direction: column;
    gap: 20px;
  }

  .bottom-section {
    width: 80vw;
  }

  .throw-error-btn {
    margin: 0 auto;
  }
}
