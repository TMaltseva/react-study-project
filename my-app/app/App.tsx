'use client';

import React from 'react';
import { wrapper } from '../store/store';
import { Provider } from 'react-redux';
import Home from './Home';

const App: React.FC = () => {
  const { store } = wrapper.useWrappedStore({});
  return (
    <Provider store={store}>
      <Home />
    </Provider>
  );
};

export default App;
