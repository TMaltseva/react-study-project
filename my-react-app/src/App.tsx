import { RouterProvider } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './store/store';
import { router } from './routes/router';

function App() {
  return (
    <Provider store={store}>
      <RouterProvider router={router} />
    </Provider>
  );
}

export default App;
