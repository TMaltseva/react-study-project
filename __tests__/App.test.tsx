import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../src/store';
import { ThemeProvider } from '../src/components/ThemeProvider';
import App from '../src/App';

test('renders App component correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ThemeProvider>
          <App />
        </ThemeProvider>
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});