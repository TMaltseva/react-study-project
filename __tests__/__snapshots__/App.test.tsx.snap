// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
<DocumentFragment>
  <main
    class="sections-wrapper"
  >
    <div
      class="top-section"
    >
      <div>
        <label
          for="search-input"
        >
          Search:
        </label>
        <input
          aria-label="Search input"
          id="search-input"
          type="text"
          value=""
        />
        <button>
          Search
        </button>
      </div>
      <button
        class="throw-error-btn"
      >
        Throw Error
      </button>
      <button
        class="theme-btn"
      >
        To dark theme
      </button>
      <div
        class="flyout "
      />
    </div>
    <div
      class="bottom-section"
    >
      <div
        class="left-section"
      >
        <p>
          Loading...
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
